### PR TITLE
Autocomplete improvements & new customization options

### PR DESCRIPTION
**Pull request by sergbass**
This PR adds extra functionality to  sonata_type_model_autocomplete, fixes few glitches & extends customization options:

1. Added callback with possibility to modify each individual item in response.
2. Removed form type check in HelperController: it made impossible to extend  	sonata_type_model_autocomplete without writing own controller with copying whole retrieveAutocompleteItemsAction method in case of extending it.
3. Added way to customize any autocomplete select2 properties in js via adding sonata_type_model_autocomplete_select2_options_js block before initializing select2 element. Reason to do this is because select2 doesn't allow modification of several options after it has been initialized, you can only destroy it and initialize again from scratch.
4. Make autocomplete item list independent of filters set in corresponding list via removing it's filters. Otherwise, if you set filter in, let's say, product list, and you're using product autocomplete elsewhere, it returns only products that match that filter, which is confusing.
5. Set error bubbling to false - prevent passing error to parent form elements, which occurs by default for compound form types.

### CODE REVIEW

**Comment by GromNaN** [comment]
The modifications are in conflict with this PR: https://github.com/sonata-project/SonataAdminBundle/pull/2454.

Could you rework your PR based on the master branch ?

File: Controller/HelperController.php [hunk 1]
```diff
@@ -396,10 +399,16 @@ public function retrieveAutocompleteItemsAction(Request $request)
                 $label = $resultMetadata->getTitle();
             }
 
-            $items[] = array(
+            $item = array(
                 'id'    => $admin->id($entity),
                 'label' => $label,
             );
+
+            if (is_callable($resultItemCallback)) {
```
Comment (pulzarraider) [inline Controller/HelperController.php]: You should throw some exception, if user will set to `response_item_callback` something, that is not callback.

File: Controller/HelperController.php [hunk 2]
```diff
@@ -396,10 +399,16 @@ public function retrieveAutocompleteItemsAction(Request $request)
...
+                $resultItemCallback($admin, $entity, $item);
```
Comment (pulzarraider) [inline Controller/HelperController.php]: Can you please use `call_user_func` to have code more consistent?

**Comment by joshlopes** [comment]
Something you guys could also re-think --> http://jsfiddle.net/X6V2s/66/

I needed to add that bit of code in a extended template to put new tags to work with separate tokens

**Comment by webdevilopers** [comment]
Have you considered adding the `group_by` functionality that is available on regular select e.g.:
bildschirmfoto vom 2015-07-29 14 21 17

**Comment by core23** [comment]
Ping @sergbass

**Comment by core23** [comment]
Are you still working on this? Please rebase @sergbass

**Pull request closed without merging**